src/Ferry.js

```diff
@@ -1,62 +1,64 @@
 'use strict';
 
-import path from 'path';
-import Storage from './Storage';
 import Router from './Router';
 import Specification from './Specification';
+import Storage from './Storage';
+
+import path from 'path';
 
 class Ferry {
+
   constructor(config) {
 
```

Comment:
I think we can get rid of all of the else statements here and throw the exceptions sooner:

```javascript
    if (typeof config === 'undefined') {
      throw new Error('Configuration is required');
    }
    
    if (!(config.specification instanceof Specification)) {
      throw new Error('Specification is required');
    }
    
    if (!(config.storage instanceof Storage)) {
      throw new Error('Storage is required');
    }
    
    if (!(config.router instanceof Router)) {
      throw new Error('Router is required');
    }

    this.specification = config.specification;
    this.specification.Ferry = this;
  
    this.storage = config.storage;
    this.storage.Ferry = this;
    
    this.router = config.router;
    this.router.Ferry = this;
    this.router.initialize(this.specification.basePath, this.specification.routes);
```